### ISSUE TITLE
Is there a way to get the transition matrix for each input token and class label pairs?

### ISSUE THREAD

**Issue by sagnik** [TeamHG-Memex/sklearn-crfsuite#3]
`sklearn-crfsuite` has a `predict_marginals_single` method which given an input sequence `x_1,x_2,..x_n` predicts the marginal probabilities of the class labels `c_1,c_2..`. In other words, it is possible to get the probability (x_k,c_i). I am guessing the marginal for a class is calculated by summing over all the transition probabilities `(c_i,c_j)` at x_k. Is there a way to get the probability for the triple `x_k,c_i,c_j`?

It is possible to get this in Mallet.

**Comment by kmike** [comment]
Hey @sagnik,

I'm not sure if this answers your question, but here is some code to get CRFsuite state and transition probabilities as numpy/scipy matrices: https://github.com/TeamHG-Memex/eli5/blob/417358f0ece0587fa18e9926756b7a9842776b72/eli5/sklearn_crfsuite/explain_weights.py#L68

**Comment by sagnik** [comment]
@kmike , thanks for your reply. I started exploring eli5, specifically, (this NER example)[http://eli5.readthedocs.io/en/latest/tutorials/sklearn_crfsuite.html#sklearn-crfsuite-tutorial ], but the statement `eli5.show_weights(crf, top=30)` gives me the error: `AttributeError: 'module' object has no attribute 'show_weights'`. Any comment on that?

**Comment by kmike** [comment]
Hey @sagnik! `eli5.show_weights` requires IPython to be installed - do you have it? 

It only makes sense to use eli5.show_weights in an IPython notebook; in a script it is better to use something like `eli5.format_as_text(eli5.explain_weights(...))`.

UnicodeDecodeError looks like an issue with https://github.com/scrapinghub/python-crfsuite library - for some reason dump of the CRF model is not encoded to utf-8. Likely there is something in the way you passed the data while training. Do you have an idea why can this happen, why can CRF get non-UTF8 data? It is still a python-crfsuite bug because a library shouldn't fail in this case.

**Comment by sagnik** [comment]
Hi @kmike , sorry for getting back so late. The encoding problem is now solved. One of the features of the CRF was the word itself and I encoded the string in UTF-8 format. However, that doesn't solve the problem I was talking about. `crf_state_coef` function gives me a matrix of the dimention n*m where n is the number of classes and m is the number of features in the training data. What I am trying to get is the transition probability between two labels when a particular token is seen during `inference`. Could you please point me to the code portion where the inference is done?
